app/src/main/java/org/simple/clinic/registration/confirmpin/RegistrationConfirmPinEffectHandler.kt

```diff
@@ -35,7 +38,7 @@ class RegistrationConfirmPinEffectHandler @AssistedInject constructor(
     return ObservableTransformer { effects ->
       effects
           .map { (pinConfirmation, entry) -> checkIfConfirmedPinIsValid(pinConfirmation, entry) }
-          .map(::PinConfirmationValidated)
+          .map { PinConfirmationValidated(it, Instant.now(utcClock)) }
```

Comment:
Can you explain how is adding timestamp is making the `Update` function pure?

Reply:
Good question! The `Instant` itself is fine to track in `Update` functions, because it is immutable. What we should not be doing is fetching the current timestamp in `Update` functions, which is why we read it in the effect handler, and pass it to `Update` as an event.